### PR TITLE
feat: 适配链接跳转的判断方式

### PR DESCRIPTION
**Pull request by GaoFuhong**
### 👀 PR includes

✨ Feature

- [x] 增加判断 meta 中 id 的方式来确定是否添加链接

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/interaction/base-interaction/click/data-cell-click.ts [hunk 1]
```diff
@@ -135,10 +135,11 @@ export class DataCellClick extends BaseEvent implements BaseEventImplement {
 
   private emitLinkFieldClickEvent(event: CanvasEvent) {
     const { cellData } = this.getCellAppendInfo(event.target);
-    const { valueField: key, data: record } = cellData;
+    const { valueField: key, id, data: record } = cellData;
 
     this.spreadsheet.emit(S2Event.GLOBAL_LINK_FIELD_JUMP, {
       key,
+      id,
```
Comment (lijinke666) [inline packages/s2-core/src/interaction/base-interaction/click/data-cell-click.ts:142]: 直接透出 cellData

File: packages/s2-core/src/interaction/base-interaction/click/row-text-click.ts [hunk 2]
```diff
@@ -24,10 +24,12 @@ export class RowTextClick extends BaseEvent implements BaseEventImplement {
 
       const { cellData } = this.getCellAppendInfo(event.target);
       const key = cellData.key;
+      const id = cellData.id;
       const rowData = this.getRowData(cellData);
 
       this.spreadsheet.emit(S2Event.GLOBAL_LINK_FIELD_JUMP, {
         key,
+        id,
```
Comment (lijinke666) [inline packages/s2-core/src/interaction/base-interaction/click/row-text-click.ts:32]: 同上

File: packages/s2-core/src/cell/row-cell.ts [hunk 3]
```diff
@@ -229,8 +229,10 @@ export class RowCell extends HeaderCell {
   protected drawLinkFieldShape() {
     const { linkFields = [] } = this.headerConfig;
     const { linkTextFill } = this.getTextStyle();
-
-    super.drawLinkFieldShape(linkFields.includes(this.meta.key), linkTextFill);
+    const isLinkField = linkFields.some(
```
Comment (lijinke666) [inline packages/s2-core/src/cell/row-cell.ts:232]: 加个测试 packages/s2-core/__tests__/spreadsheet/row-link-spec.ts

Comment (GaoFuhong) [reply packages/s2-core/src/cell/row-cell.ts:232]: 已补充。

Comment (GaoFuhong) [reply packages/s2-core/src/cell/row-cell.ts:232]: done

File: packages/s2-core/src/interaction/base-interaction/click/data-cell-click.ts (hunk 1 above)
Comment (GaoFuhong) [reply packages/s2-core/src/interaction/base-interaction/click/data-cell-click.ts]: done

**Review by lijinke666 — APPROVED** [review]

File: s2-site/docs/api/components/sheet-component.zh.md [hunk 4]
```diff
@@ -95,7 +95,7 @@ order: 0
 | onMouseUp | 表格鼠标松开事件 | (event: CanvasEvent) => void |  |  |
 | onSelected | 单元格选中事件 | (cells: ( [Cell](/docs/api/basic-class/base-cell)[] ) => void |  |  |
 | onReset | 交互状态重置事件 | (event: KeyboardEvent) => void |  |  |
-| onLinkFieldJump | 链接字段跳转事件 | (data: { key: string; record: [Data](/docs/api/general/S2DataConfig#data) }) => void |  |  |
+| onLinkFieldJump | 链接字段跳转事件 | (data: { key: string; cellData: [Node](/docs/api/basic-class/node); record: [Data](/docs/api/general/S2DataConfig#data) }) => void |  |  |
```
Comment (lijinke666) [inline s2-site/docs/api/components/sheet-component.zh.md:98]: 下面 vue 还有一个

Comment (lijinke666) [reply s2-site/docs/api/components/sheet-component.zh.md:98]:
```diff
- 链接字段跳转事件
+ 链接字段跳转事件 (cellData: @antv/s2 1.37.0 新增)
```

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**